GVFS/GVFS.Platform.Mac/MacPlatform.cs

```diff
@@ -159,5 +159,11 @@ public override bool IsGitStatusCacheSupported()
         {
             return new MacFileBasedLock(fileSystem, tracer, lockPath);
         }
+
+        public override bool KillProcessTree(int processId)
+        {
+            ProcessResult result = ProcessHelper.Run("kill", $"{processId}");

```

Comment:
How did you test this on Mac?

Reply:
I left a comment on top saying "Testing in progress for Mac".  I actually had to switch to "pkill -P" to make this properly work.  

To answer your question though I switched the background job to push a large file (so it would take a while) and then ensured the push was killed on unmount.  Push spawns several children so it's a good test case.